jdbc-core/src/main/java/com/salesforce/datacloud/jdbc/core/DataCloudConnection.java

```diff
@@ -284,19 +285,23 @@ public DataCloudResultSet getChunkBasedResultSet(String queryId, long chunkId) t
     }
 
     public ResultSetMetaData getSchemaForQueryId(String queryId) throws SQLException {
-        val executor = HyperGrpcClientExecutor.forSubmittedQuery(getStub());
-        val infos = executor.getQuerySchema(queryId);
+        HyperGrpcClientExecutor executor = HyperGrpcClientExecutor.forSubmittedQuery(getStub());
+        Iterator<QueryInfo> infos = executor.getQuerySchema(queryId);
 
         try {
-            Schema schema = deserializeMessage(getByteBuffer(queryId, infos));
+            Iterator<ByteString> byteStringIterator = ProtocolMappers.fromQueryInfo(infos);
+            if (!byteStringIterator.hasNext()) {
+                throw new DataCloudJDBCException("No schema data available for queryId: " + queryId);
+            }
```

Comment:
Do we have a guarantee in our interface that the schema will always be the first message returned? If so, we might not even need an `byteStringIterator` here?

Reply:
No, currently the query status is sent first, the schema is sent after that.

Reply:
ok, then we can't exploit that to simplify the code. The code as written is fine by me, then